pbrd/pbr_vty.c

```diff
@@ -127,10 +127,13 @@ DEFPY(pbr_map_match_src, pbr_map_match_src_cmd,
 	pbrms->family = prefix->family;
 
 	if (!no) {
-		if (prefix_same(pbrms->src, prefix))
+		if (pbrms->src && prefix_same(pbrms->src, prefix))
 			return CMD_SUCCESS;
-
-		if (!pbrms->src)
+		else if (pbrms->src) {
```

Comment:
Instead of making a redundant check here why not nest prefix_same() under f (pbrms->src).  That improves code readability a bit.